src/validation.cpp

```diff
@@ -2551,6 +2553,9 @@ bool CChainState::ActivateBestChainStep(CValidationState& state, const CChainPar
             // This is likely a fatal error, but keep the mempool consistent,
             // just in case. Only remove from the mempool in this case.
             UpdateMempoolForReorg(disconnectpool, false);
+            if (!g_abort_node) {
```

Comment:
Wouldn't it make more sense to move this check into `AbortNode` otherwise it seems that this is a flag to indicate the node should be aborted as opposed to a flag that indicates the nodes has been aborted via `AbortNode`.

Reply:
I don't think it is worth the burden of having this undocumented global in the source code to ease today's review of the diff at the cost of all future review of the overall code.

I'd prefer to be consistent by allowing AbortNode to be called multiple times and then either overwriting the warning string or keeping the previous one, but special casing the reorg warning to not overwrite any previous warnings seems arbitrary.

Reply:
Perhaps I should get rid of the call to `AbortNode` in DisconnectTip, and then do away with this new global and unconditionally call AbortNode in ABCStep?  That would result in a behavior change for invalidateblock and certain failures in RewindBlockIndex, but that might actually be desirable anyway.